projects/plugins/jetpack/extensions/blocks/subscriptions/class-jetpack-subscription-site.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Adds support for Jetpack Subscription Site feature.
+ *
+ * @package automattic/jetpack
+ * @since $$next-version$$
+ */
+
+namespace Automattic\Jetpack\Extensions\Subscriptions;
+
+/**
+ * Jetpack_Subscription_Site class.
+ */
+class Jetpack_Subscription_Site {
+	/**
+	 * Jetpack_Subscription_Site singleton instance.
+	 *
+	 * @var Jetpack_Subscription_Site|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscription_Site instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscription_Site();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Handles Subscribe block placements.
+	 *
+	 * @return void
+	 */
+	public function handle_subscribe_block_placements() {
+		if ( ! $this->is_subscription_site_feature_enabled() ) {
+			return;
+		}
+
+		$this->handle_subscribe_block_post_end_placement();
+	}
+
+	/**
+	 * Returns true if Subscription Site feature is enabled.
+	 *
+	 * @return bool
+	 */
+	protected function is_subscription_site_feature_enabled() {
+		return (bool) apply_filters( 'jetpack_subscription_site_enabled', false );
```

Comment:
You'll want to add a docblock for this filter, so the codex parser can pick it up.

Reply:
Not sure why we need it but added. This is a feature flag and we don't want to expose it.

Reply:
@jeherve fine to keep the filter internal? We wouldn't want to use filter deprecation when removing it.

Reply:
I suppose that would be okay. I guess I'm just a bit weary of filters that are temporary but that end up staying for some time. :) 

Let's maybe add a short comment saying that the filter is temporary?

Reply:
Added :)